tools/topology/topology2/include/dais/mic_config.conf

```diff
@@ -0,0 +1,126 @@
+# mic vendor config for nhlt type mic config
+#
+#	Object.Base.vendor_mic_config."0" {
+#		mic_type			0
+#		location			4
+#		speaker_position_distance	0
+#		horizontal_offset		0
+#		vertical_offset			0
+#		frequency_low_band		0
+#		frequency_high_band		0
+#		direction_angle			0
+#		elevation_angle			0
+#		vertical_angle_begin		0
+#		vertical_angle_end		0
+#		horizontal_angle_begin		0
+#		horizontal_angle_end		0
+#	}
+#
+#	Object.Base.vendor_mic_config."1" {
+#		mic_type			0
+#		location			4
+#		speaker_position_distance	0
+#		horizontal_offset		0
+#		vertical_offset			0
+#		frequency_low_band		0
+#		frequency_high_band		0
+#		direction_angle			0
+#		elevation_angle			0
+#		vertical_angle_begin		0
+#		vertical_angle_end		0
+#		horizontal_angle_begin		0
+#		horizontal_angle_end		0
+#	}
+#
+Class.Base."vendor_mic_config" {
+
+	DefineAttribute."mic_id" {}
+
+	# 0 Omnidirectional
+	# 1 Subcardioid
+	# 2 Cardioid
+	# 3 Supercardioid
+	# 4 Hypercardioid
+	# 5 Figure-8
+	# 6 Reserved
+	# 7 Vendordefined
+	DefineAttribute."mic_type" {}
+
+	# 0 Top
+	# 1 Bottom
+	# 2 Left
+	# 3 Right
+	# 4 Front (default)
+	# 5 Rear
+	DefineAttribute."location" {}
+
+	# in mm
+	DefineAttribute."speaker_position_distance" {}
+
+	# in mm
+	DefineAttribute."horizontal_offset" {}
+
+	# in mm
+	DefineAttribute."vertical_offset" {}
+
+	# 5 x Hz
+	DefineAttribute."frequency_low_band" {}
+
+	# 500 x Hz
+	DefineAttribute."frequency_low_band" {}
+
+	# -180 - + 180
+	DefineAttribute."direction_angle" {}
+
+	# -180 - + 180
+	DefineAttribute."elevation_angle" {}
+
+	# -180 - + 180 with 2 deg step
+	DefineAttribute."vertical_angle_begin" {}
+
+	# -180 - + 180 with 2 deg step
+	DefineAttribute."vertical_angle_end" {}
+
+	# -180 - + 180 with 2 deg step
+	DefineAttribute."horizontal_angle_begin" {}
+
+	# -180 - + 180 with 2 deg step
+	DefineAttribute."horizontal_angle_end" {}
+
+	attributes {
+		constructor [
```

Comment:
@juimonen please add "!" to override array elems in case these conf files are include multiple times

Reply:
ok